launch/model_endpoint.py

```diff
@@ -216,8 +218,30 @@ def get(self) -> EndpointResponse:
 class Endpoint:
     """An abstract class that represent any kind of endpoints in Scale Launch"""
 
-    def __init__(self, model_endpoint: ModelEndpoint):
+    def __init__(self, model_endpoint: ModelEndpoint, client):
         self.model_endpoint = model_endpoint
+        self.client = client
+
+    def _update_model_endpoint(self):
+        resp = self.client.connection.get(
+            os.path.join(ENDPOINT_PATH, self.model_endpoint.name)
```

Comment:
Can we add unit tests? Basically, we'd mock out the connection to the server, return some pre-defined mock objects, and verify that `status()` returns the updated value.

Reply:
done!